drkafka/src/test/java/com/pinterest/doctorkafka/KafkaClusterTest.java

```diff
@@ -0,0 +1,117 @@
+package com.pinterest.doctorkafka;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.pinterest.doctorkafka.config.DoctorKafkaClusterConfig;
+import com.pinterest.doctorkafka.config.DoctorKafkaConfig;
+import com.pinterest.doctorkafka.replicastats.ReplicaStatsManager;
+import com.pinterest.doctorkafka.util.OutOfSyncReplica;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.UniformReservoir;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.PriorityQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+class KafkaClusterTest {
+  private static final String TOPIC = "test_topic";
+  private static final String ZOOKEEPER_URL = "zk001/cluster1";
+
+  @Test
+  void getAlternativeBrokersDuplicateReassignmentTest() throws Exception{
```

Comment:
Could you add comments to explain the logic behind this test?  That will be helpful for other readers.